docs/index.rst

```diff
@@ -438,7 +451,241 @@ Permissions
+
+  .. sourcecode:: http
+
+     HTTP/1.1 200 OK
+     Content-Type: application/json
+
+     [{
+       "id": "0a2d19e0-bb10-4b84-98cc-52a82b6ed427",
+       "url": "/users/1",
+       "email": "foo@bar.org",
+       "first_name": "Joan",
+       "last_name": "Watson"
+     }]
+
+    :query number page:
+      1-based index of the page of users to show. Defaults to ``1``.
+
+    :query number per_page:
+      Number of users to show per page. Defaults to ``30``. Maximum is ``100``.
+
+.. http:get:: /users/(str:user_id)
+
+  Retrieves the :ref:`description <data-users>` of the user with id
+  ``user_id``.
+
+  .. sourcecode:: http
+
+     GET /users/0a2d19e0-bb10-4b84-98cc-52a82b6ed427 HTTP/1.1
+
+  .. sourcecode:: http
+
+     HTTP/1.1 200 OK
+     Content-Type: application/json
+
+     {
+       "id": "0a2d19e0-bb10-4b84-98cc-52a82b6ed427",
+       "url": "/users/0a2d19e0-bb10-4b84-98cc-52a82b6ed427",
+       "email": "foo@bar.org",
+       "first_name": "Joan",
+       "last_name": "Watson"
+     }
+
+.. http:put:: /users/(str:user_id)
+
+  Replaces the :ref:`description <data-users>` of the user with id ``user_id``
+  with the description given in the request body and returns the given
+  description, along with the user's ``id`` and the ``url`` for accessing the
+  user's description.
+
+  This operation is only accessible to the authenticated user if their user id is ``user_id``, or if they have :ref:`admin permission <permissions-admin>`.
+
+  .. sourcecode:: http
+
+     PUT /users/0a2d19e0-bb10-4b84-98cc-52a82b6ed427 HTTP/1.1
+     Content-Type: application/json
+
+     {
+       "first_name": "Joan",
+       "last_name": "Watson"
+     }
+
+  .. sourcecode:: http
+
+     HTTP/1.1 200 OK
+     Content-Type: application/json
+
+     {
+       "id": "0a2d19e0-bb10-4b84-98cc-52a82b6ed427",
+       "url": "/users/0a2d19e0-bb10-4b84-98cc-52a82b6ed427",
+       "email": "foo@bar.org",
+       "first_name": "Joan",
+       "last_name": "Watson"
+     }
+
+.. http:patch:: /users/(str:user_id)
+
+  Partially updates the :ref:`description <data-users>` of the user with id
+  ``user_id`` using the :ref:`instructions <overview-partial-updates>` given in
+  the request body and returns the given user's description, along with the
+  user's ``id`` and the ``url`` for accessing the user's description.
+
+  This operation is only accessible to the authenticated user if their user id is ``user_id``, or if they have :ref:`admin permission <permissions-admin>`.
+
+  .. sourcecode:: http
+
+     PATCH /users/0a2d19e0-bb10-4b84-98cc-52a82b6ed427 HTTP/1.1
+     Content-Type: application/json-patch+json
+
+     [{
+       "first_name": "Joan"
+     }]
+
+  .. sourcecode:: http
+
+     HTTP/1.1 200 OK
+     Content-Type: application/json
+
+     {
+       "id": "0a2d19e0-bb10-4b84-98cc-52a82b6ed427",
+       "url": "/users/0a2d19e0-bb10-4b84-98cc-52a82b6ed427",
+       "email": "foo@bar.org",
+       "first_name": "Joan",
+       "last_name": "Watson"
+     }
```

Comment:
Intentionally left out support for creating users and setting/changing authentication details, seems better for those to be done via the user authentication system we end up using (django?) directly, rather than having this happen via the API.

Thoughts?